presto-main/src/main/java/com/facebook/presto/type/DecimalCasts.java

```diff
@@ -148,6 +148,11 @@ private static SqlScalarFunction castFunctionFromDecimalTo(TypeSignature to, Str
                             else {
                                 tenToScale = bigIntegerTenToNth(intScale(scale));
                             }
+
+                            if (context.containsLiteral("x")) {
```

Comment:
You assume that `x` is text type length (convention based). Maybe pass it as argument:
`Optional<String> lengthLiteral`
then you don't have to rely on `x` convention name and you wouldn't need to check if context contains literal.

Reply:
I'm sorry, but couldn't understand this completely. Do you mean that `Optional<String> lengthLiteral` should be added to `castFunctionFromDecimalTo` function's argument?